ampd/src/handlers/config.rs

```diff
@@ -10,14 +12,25 @@ use crate::evm::finalizer::Finalization;
 use crate::types::TMAddress;
 use crate::url::Url;
 
-#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[derive(Clone, Deserialize, Serialize, PartialEq)]
 pub struct Chain {
     pub name: ChainName,
     pub rpc_url: Url,
     #[serde(default)]
     pub finalization: Finalization,
 }
 
+impl Debug for Chain {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let rpc_url = "redacted".to_string();
+        f.debug_struct("Chain")
+            .field("name", &self.name)
+            .field("rpc_url", &rpc_url)
+            .field("finalization", &self.finalization)
+            .finish()
+    }
```

Comment:
```suggestion
        f.debug_struct("Chain")
            .field("name", &self.name)
            .field("rpc_url", "redacted")
            .field("finalization", &self.finalization)
            .finish()
    }
```
or go with the suggestion to make it part of the url definition